examples/extending/extending-leaflet.md

```diff
@@ -0,0 +1,445 @@
+	* Adding new methods and options
+	* Changing some methods
+	* Using ``
+* Changing parts of an existing class (replacing how a class method works)
+
+This tutorial covers some classes and methods available only in Leaflet 1.0.0. Use caution if you are developing a plugin compatible for previous versions.
+
+## `L.Class`
+
+Javascript is a bit of a weird language. It's not really an object-oriented language, but rather a [prototype-oriented language](https://en.wikipedia.org/wiki/Prototype-based_programming). This has made JavaScript historically difficult to use class inheritance in the classic OOP meaning of the term.
+
+Leaflet works around this by having `L.Class`, which eases up class inheritance.
+
+### `L.Class.extend()`
+
+In order to create a subclass of anything in Leaflet, use the `.extend()` method. This accepts one parameter: a plain object with key-value pairs, each key being the name of a property or method, and each value being the initial value of a property, or the implementation of a method:
+
+    var MyDemoClass = L.Class.extend({
+    
+        // A property with initial value = 42
+        myDemoProperty: 42,   
+    
+        // A method 
+        myDemoMethod: function() { return this.myDemoProperty; }
+        
+    });
+
+    var myDemoInstance = new MyDemoClass();
+    
+    // This will output "42" to the development console
+    console.log( myDemoInstance.myDemoMethod() );   
+
+
+When naming classes, methods and properties, adhere to the following conventions:
+    
+* Function, method, property and factory names should be in [`lowerCamelCase`](https://en.wikipedia.org/wiki/CamelCase).
+* Class names should be in [`UpperCamelCase`](https://en.wikipedia.org/wiki/CamelCase).
+* Private properties and methods start with an underscore (`_`). This doesn't make them private: just tells other programmers that they should stay away from them.
+
+### `L.Class.include()`    
+    
+If a class is already defined, existing properties/methods can be redefined, or new ones can be added by using `.include()`:
+
+    MyDemoClass.include({
+    
+        // Adding a new property to the class
+        _myPrivateProperty: 78,
+        
+        // Redefining a method
+        myDemoMethod: function() { return this._myPrivateProperty; }
+    
+    });
+
+    var mySecondDemoInstance = new MyDemoClass();
+    
+    // This will output "50"
+    console.log( mySecondDemoInstance.myDemoMethod() );
+    
+    // However, properties and methods from before still exist
+    // This will output "42"
+    console.log( mySecondDemoInstance.myDemoProperty );
+
+### `L.Class.initialize()`
+    
+In OOP, classes have a constructor method. In Leaflet's `L.Class`, the constructor method is always named `initialize`.
+
+If your class has some specific `options`, it's a good idea to initialize them with `L.setOptions()` in the constructor. This utility function will merge the provided options with the default options of the class.
```

Comment:
I'm thinking of more or less switching the order of this paragraph and the next, and maybe just add a few words about what `options` actually are.